streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsKafkaClient.java

```diff
@@ -0,0 +1,216 @@
+
+        final RequestSend send = new RequestSend(brokerId,
+                kafkaClient.nextRequestHeader(apiKeys),
+                request);
+
+        final ClientRequest clientRequest = new ClientRequest(systemTime.milliseconds(), true, send, null);
+
+
+        final long readyTimeout = systemTime.milliseconds() + MAX_WAIT_TIME_MS;
+        while (!kafkaClient.ready(brokerNode, systemTime.milliseconds()) && systemTime.milliseconds() < readyTimeout) {
+            kafkaClient.poll(streamsConfig.getLong(StreamsConfig.POLL_MS_CONFIG), systemTime.milliseconds());
+        }
+        if (!kafkaClient.ready(brokerNode, systemTime.milliseconds())) {
+            throw new StreamsException("Timed out waiting for node=" + brokerNode + " to become available");
+        }
+        kafkaClient.send(clientRequest, systemTime.milliseconds());
+
+        final long responseTimeout = systemTime.milliseconds() + MAX_WAIT_TIME_MS;
+        // Poll for the response.
+        while (systemTime.milliseconds() < responseTimeout) {
+            List<ClientResponse> responseList = kafkaClient.poll(streamsConfig.getLong(StreamsConfig.POLL_MS_CONFIG), systemTime.milliseconds());
+            for (ClientResponse clientResponse: responseList) {
+                if (clientResponse.request().equals(clientRequest)) {
+                    return clientResponse;
+                }
+            }
+        }
+        throw new StreamsException("Failed to get response from node=" + brokerNode + " within timeout");
+    }
+
+
+    /**
+     * Get the metadata for a topic.
+     * @param topic
+     * @return
+     */
+    public MetadataResponse.TopicMetadata getTopicMetadata(final String topic) {
+        final MetadataRequest metadataRequest = new MetadataRequest(Collections.singletonList(topic));
+
+        final ClientResponse clientResponse = sendRequest(metadataRequest.toStruct(), ApiKeys.METADATA);
+        final MetadataResponse metadataResponse = new MetadataResponse(clientResponse.responseBody());
+
+        for (MetadataResponse.TopicMetadata topicMetadata: metadataResponse.topicMetadata()) {
+            if (topicMetadata.topic().equalsIgnoreCase(topic)) {
+                return topicMetadata;
+            }
+        }
+
+        throw new StreamsException("Topic " + topic + " was not found.");
+    }
+
+    /**
+     * Check to see if a topic exists.
+     * @param topicName
+     * @return
+     */
+    public boolean topicExists(final String topicName) {
+
+        final Properties props = new Properties();
+        props.put(CommonClientConfigs.BOOTSTRAP_SERVERS_CONFIG, streamsConfig.getList(StreamsConfig.BOOTSTRAP_SERVERS_CONFIG));
+        props.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, Serdes.String().deserializer().getClass());
+        props.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, Serdes.String().deserializer().getClass());
+
+        final KafkaConsumer<String, String> consumer = new KafkaConsumer<>(props);
```

Comment:
Sorry I meant `MetadataRequest`, not `ListOffsetRequest`.

If you specify the topics to be the empty list (i.e. `ALL_TOPICS_REQUEST`), then the broker will not create any topics even with `auto.topic.create` is true.